webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx

```diff
@@ -0,0 +1,158 @@
+    }
+  }, [ group, form ]);
+  useEffect(() => {
+    fetchGroup();
+  }, [ fetchGroup ]);
+
+  return (
+    <Form
+      form={form}
+      labelCol={{ span: 8 }}
+      wrapperCol={{ span: 14 }}>
+      <Form.Item
+        label={GROUP_NAME_LABEL}
+        name={GROUP_NAME_NAME}
+        required
+        rules={[
+          {
+            message: 'Please type in your group name.',
+            required: true,
+          },
+        ]}
+        validateTrigger={[ 'onSubmit' ]}>
+        <Input autoFocus maxLength={128} placeholder="Group Name" />
+      </Form.Item>
+      {group ? (
+        <Form.Item
+          label={USER_ADD_LABEL}
+          name={USER_ADD_NAME}>
+          <Select
+            loading={isLoading}
+            mode="multiple"
+            optionFilterProp="children"
+            placeholder="Add Users"
+            showSearch>{
+              users.filter(
+                (u) => !groupDetail?.users?.map((gu) => gu.id).includes(u.id),
+              ).map((u) => (
+                <Select.Option key={u.id} value={u.id}>{u.displayName || u.username}</Select.Option>
+              ))
+            }
+          </Select>
+        </Form.Item>
+      ) : (
+        <Form.Item
+          label={USER_LABEL}
+          name={USER_ADD_NAME}>
+          <Select mode="multiple" optionFilterProp="children" placeholder="Add Users" showSearch>{
+            users.map((u) => (
+              <Select.Option key={u.id} value={u.id}>{u.displayName || u.username}</Select.Option>
+            ))
+          }
+          </Select>
+        </Form.Item>
+      )}
+    </Form>
+  );
+};
+
+interface ModalProps {
+  group?: V1GroupSearchResult;
+  onClose?: () => void;
+  users: DetailedUser[];
+}
+
+const useModalCreateGroup = ({ onClose, users, group }: ModalProps): ModalHooks => {
+
+  const [ form ] = Form.useForm();
+
+  const { modalOpen: openOrUpdate, ...modalHook } = useModal();
+
+  const handleCancel = useCallback(() => {
+    form.resetFields();
+  }, [ form ]);
+
+  const handleOkay = useCallback(async () => {
+    await form.validateFields();
+
+    try {
+      const formData = form.getFieldsValue();
+      if (group) {
+        await updateGroup({ groupId: group.group.groupId, ...formData });
+        message.success(API_SUCCESS_MESSAGE_EDIT);
+      } else {
+        await createGroup(formData);
+        message.success(API_SUCCESS_MESSAGE_CREATE);
+      }
+      form.resetFields();
+      onClose?.();
+    } catch (e) {
+      message.error('error creating new group');
+      handleError(e, { silent: true, type: ErrorType.Input });
+
+      // Re-throw error to prevent modal from getting dismissed.
+      throw e;
+    }
+  }, [ form, onClose, group ]);
+
+  const modalOpen = useCallback(() => {
+    openOrUpdate({
+      closable: true,
+      content: <ModalForm form={form} group={group} users={users} />,
+      icon: null,
+      okText: group ? MODAL_HEADER_LABEL_EDIT : MODAL_HEADER_LABEL_CREATE,
```

Comment:
nit: i'd prefer something like "Save Group" and/or "Save Changes" for the button, but reusing the modal title is fine too, really.